com.unity.multiplayer.mlapi/Runtime/Serialization/BitReader.cs

```diff
@@ -393,7 +393,10 @@ public Quaternion ReadRotationPacked()
             float y = ReadSinglePacked();
             float z = ReadSinglePacked();
 
-            float w = Mathf.Sqrt(1f - Mathf.Pow(x, 2) - Mathf.Pow(y, 2) - Mathf.Pow(z, 2));
+            // numerical precision issues can make the remainder very slightly negative.
+            // In this case, use 0 for w as, otherwise, w would be NaN.
+            float remainder = 1f - Mathf.Pow(x, 2) - Mathf.Pow(y, 2) - Mathf.Pow(z, 2);
```

Comment:
I think I'd recommend instead doing just:

`float remainder = 1f - x*x - y*y  - z*z`;

Because maybe the compiler is smart enough to devolve these, maybe it isn't...

Reply:
yeah, very valid point. But this got merged already. I'll set it aside and check other instances of `sqrt` and `pow`, for a later PR